onnx/defs/nn/defs.cc

```diff
@@ -3279,13 +3314,26 @@ ONNX_OPERATOR_SET_SCHEMA(
 
           // Combine rotated parts with non-rotated parts
           builder.Add("XConcat = Concat <axis = -1> (XRotated, XNoRotate)");
-          if (num_heads > 0) {
-            builder.Add("YTransposed = Identity(XConcat)");
-          } else {
-            builder.Add("YTransposed = Transpose <perm = [0, 2, 1, 3]> (XConcat)");
-          }
-          // Reshape back to 3D shape if input is a 3D tensor
-          builder.Add("XShape = Shape(X)").Add("Y = Reshape(YTransposed, XShape)");
+
+          // Select the correct output based on number of dimensions
+          // If input is 3D, reshape the output to 3D
+          // If input is 4D, transpose the output to match original shape
+          builder.Add("YCond = Equal(Rank, Three1D)");
```

Comment:
Shouldn't this be the same condition as the earlier one? Why not just reuse the same?